01-messaging.md

```diff
@@ -119,24 +119,25 @@ For simplicity of diagnosis, it is often useful to tell the peer that something
 
 1. type: 17 (`error`)
 2. data:
-   * [8:channel-id]
+   * [32:channel-id]
    * [2:len]
    * [len:data]
 
 The 2-byte `len` field indicates the number of bytes in the immediately following field.
 
 #### Requirements
 
+The channel is referred to by `channel-id` unless `channel-id` is all-zero, in which case it refers to all channels.
+
 A node SHOULD send `error` for protocol violations or internal
 errors which make channels unusable or further communication unusable.
 A node MAY send an empty [data] field.  A node sending `error` MUST
-fail the channel referred to by the `channel-id`, or if `channel-id`
-is `0xFFFFFFFFFFFFFFFF` it MUST fail all channels and MUST close the connection.
+fail the channel referred to by the error message, or if `channel-id` is all-zero, MUST
+fail all channels and MUST close the connection.
 A node MUST set `len` equal to the length of `data`.  A node SHOULD include the raw, hex-encoded transaction in reply to a `funding_created`, `funding_signed`, `closing_signed` or `commitment_signed` message when failure was caused by an invalid signature check.
 
-A node receiving `error` MUST fail the channel referred to by
-`channel-id`, or if `channel-id` is `0xFFFFFFFFFFFFFFFF` it MUST fail
-all channels and MUST close the connection.  A receiving node MUST truncate
+A node receiving `error` MUST fail the channel referred to by the message,
+or if `channel-id` is all-zero, it MUST fail all channels and MUST close the connection.  If no existing channel is referred to by the message, the receiver MUST ignore the message. A receiving node MUST truncate
```

Comment:
Ack. I've made that clear, by using "zero" instead of "all-zero" with the first usage followed by " (ie. all bytes zero)".  That's explicit, though it's hard to think of any other meaning of zero :)